plugin.js

```diff
@@ -8,6 +8,20 @@ const { Signer, sign, unsign } = require('./signer')
 const kReplySetCookies = Symbol('fastify.reply.setCookies')
 const kReplySetCookiesHookRan = Symbol('fastify.reply.setCookiesHookRan')
 
+function fastifyCookieGetCookie (name) {
```

Comment:
Why introduce a completely different API on `reply` than exists on `request`?

Reply:
To be certain, do you mean why not something like `reply.cookies.name`? If so, because the cookies aren't stored in an Object but a Map, we can't use a simple object, also, we have to do a search so it can't be as simple as a property access. It kinda mirrors `setCookie`

If you instead mean why I didn't decorate the request, it felt like, since these are the cookies to be sent back to the client, reply was appropriate

Reply:
I mean that cookies on the request are accessible from `request.cookies['some-cookie']`. I think that the API on the reply should be consistent. As for the storage mechanism:

```js
const map = new Map()
map._set = map.set
map.set = function set(name, value) {
	map._set.call(this, name, value)
	Object.defineProperty(map, name, { value })
}
```

Or several other methods of a custom `Map` impelementation.

> also, we have to do a search so it can't be as simple as a property access.

I don't follow.

Reply:
I will take a look at this, setCookie(name, value) could also just set the resulting cookie object as a property of reply.cookies

However, that would make Map redundant if I'm not mistaken and cost us in terms of performance (from Map to Object in dictionary mode)